pkg/invariants/incluster_disruption_serializer/remote.go

```diff
@@ -381,18 +255,177 @@ func fetchEventsFromFileOnNode(ctx context.Context, clientset *kubernetes.Client
 	if err != nil {
 		return filteredEvents, fmt.Errorf("failed to fetch file %s on node %s: %v", filePath, nodeName, err)
 	}
-	allEvents, err := monitorserialization.EventsFromJSON(allBytes)
+	events, err := monitorserialization.EventsFromJSON(allBytes)
 	if err != nil {
 		return nil, fmt.Errorf("failed to convert file %s from node %s to intervals: %v", filePath, nodeName, err)
 	}
-	fmt.Fprintf(os.Stdout, "Fetched %d events from node %s\n", len(allEvents), nodeName)
-	// Keep only disruption events
-	for _, event := range allEvents {
-		if !strings.HasPrefix(event.Locator, "disruption/") {
+	fmt.Fprintf(os.Stderr, "Fetched %d events from node %s\n", len(events), nodeName)
+	return events, err
+}
+
+type InvariantInClusterDisruption struct {
+	adminRESTConfig *rest.Config
+}
+
+func NewInvariantInClusterDisruption() invariants.InvariantTest {
+	return &InvariantInClusterDisruption{}
+}
+
+func (i *InvariantInClusterDisruption) StartCollection(ctx context.Context, adminRESTConfig *rest.Config, _ monitorapi.RecorderWriter) error {
+	fmt.Fprintf(os.Stderr, "Starting in-cluster monitoring daemonsets\n")
+	i.adminRESTConfig = adminRESTConfig
+	kubeClient, err := kubernetes.NewForConfig(i.adminRESTConfig)
+	if err != nil {
+		return err
+	}
+	configClient, err := configclient.NewForConfig(i.adminRESTConfig)
+	if err != nil {
+		return err
+	}
+	infra, err := configClient.ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	internalAPI, err := url.Parse(infra.Status.APIServerInternalURL)
+	if err != nil {
+		return err
+	}
+	apiIntHost := internalAPI.Hostname()
+
+	err = createNamespace(ctx, kubeClient)
+	if err != nil {
+		return err
+	}
+	err = createServiceAccount(ctx, kubeClient)
+	if err != nil {
+		return err
+	}
+	err = createRBACPrivileged(ctx, kubeClient)
+	if err != nil {
+		return err
+	}
+	err = createMonitorRole(ctx, kubeClient)
+	if err != nil {
+		return err
+	}
+	err = createMonitorCRB(ctx, kubeClient)
+	if err != nil {
+		return err
+	}
+	err = createServiceNetworkDS(ctx, kubeClient)
```

Comment:
daemonsets aren't drained.  I bet you want a thing that will drain during node shutdowns.

Reply:
I don't know of a way to make it drain during `oc adm drain` or MCD node upgrade. However, we'll have them dump data on disk when the node is rebooted anyway

Reply:
deployment with anti-affinity.